### DISCUSSION TITLE
Where do you set the .ipa filename?

### DISCUSSION THREAD

**Discussion by ji-denizturgut** [fastlane/fastlane#17717] [Q&A]
I have two targets with 3 schemes and the names are all different. When I do `fastlane build` the `.ipa` filename was incorrect. The file was using target instead of scheme so I had to manually update the filename in order to upload it to TestFlight. Can anyone tell me which line of scrip is setting the .ipa file? Thank you.

Fastfile

```
default_platform(:ios)
ENV["FASTLANE_APPLE_APPLICATION_SPECIFIC_PASSWORD"] = "password"

  desc "Create ipa"
  lane :build do
    increment_build_number_in_plist
    gym
  end

 desc "Upload to App TestFlight"
  lane :upload do
    upload_to_testflight(
    	app_identifier: ENV['BUNDLE_IDENTIFIER'],
    	ipa: "./fastlane/builds/#{ENV['SCHEME']}.ipa",
      skip_waiting_for_build_processing: true
    	)
  end

  desc "Create app ipa and upload to TestFlight"
  lane :do_everything do
    if ENV['SCHEME'] != nil
      build
      upload
    else
      execute_for_all_envs{ do_everything }
    end

  end

  def execute_for_all_envs
    # 1
    schemeList = Dir.glob(".env.*")
    # 2
    schemeList.each do |file|
      # 3
      Dotenv.overload(file)
      # 4
      yield
    end
  end
```

**Comment by rogerluan** [comment]
Hey @ji-denizturgut 👋 

Looks like this is the code you're looking for, for your very specific case (i.e. an iOS app): 
https://github.com/fastlane/fastlane/blob/e89cdcfb869ad0640438cc5f6ed261bb50887641/gym/lib/gym/runner.rb#L234-L241
